tests/validate_tests.rs

```diff
@@ -0,0 +1,512 @@
+
+    let mut tx = TypedTransaction::default();
+    tx.set_to(factory_address);
+    tx.set_data(init_func);
+
+    let call_init_code_for_addr = context.client.call(&tx, None).await?;
+
+    let (head, address) = call_init_code_for_addr.split_at(12);
+    anyhow::ensure!(
+        !head.iter().any(|i| *i != 0),
+        format!(
+            "call init code returns non address data : {:?}",
+            call_init_code_for_addr
+        )
+    );
+
+    let sender = Address::from_slice(address);
+
+    Ok(UserOperation {
+        sender,
+        nonce: U256::zero(),
+        init_code,
+        call_data: Bytes::default(),
+        call_gas_limit: U256::from(1000000),
+        verification_gas_limit: U256::from(1000000),
+        pre_verification_gas: U256::from(50000),
+        max_fee_per_gas: U256::from(0),
+        max_priority_fee_per_gas: U256::from(0),
+        paymaster_and_data,
+        signature,
+    })
+}
+
+async fn existing_storage_account_user_op(validate_rule: String, pm_rule: String) -> UserOperation {
+    let context = get_global_context().await;
+
+    let mut paymaster_and_data = vec![];
+    paymaster_and_data.extend_from_slice(context.paymaster.address.as_bytes());
+    paymaster_and_data.extend_from_slice(pm_rule.as_bytes());
+
+    let signature = Bytes::from(validate_rule.as_bytes().to_vec());
+    UserOperation {
+        sender: context.storage_account.address,
+        nonce: U256::zero(),
+        init_code: Bytes::default(),
+        call_data: Bytes::default(),
+        call_gas_limit: U256::from(1000000),
+        verification_gas_limit: U256::from(1000000),
+        pre_verification_gas: U256::from(50000),
+        max_fee_per_gas: U256::from(0),
+        max_priority_fee_per_gas: U256::from(0),
+        paymaster_and_data: Bytes::from(paymaster_and_data),
+        signature,
+    }
+}
+
+fn validate(_user_op: UserOperation) -> anyhow::Result<()> {
+    // TODO
+    Ok(())
+}
+
+async fn test_user_op(
+    validate_rule: String,
+    pm_rule: Option<String>,
+    init_code: Bytes,
+    init_func: Bytes,
+    factory_address: Address,
+) -> anyhow::Result<()> {
+    let user_op = create_test_user_op(
+        validate_rule,
+        pm_rule,
+        init_code,
+        init_func,
+        factory_address,
+    )
+    .await?;
+    validate(user_op)
+}
+
+async fn test_existing_user_op(validate_rule: String, pm_rule: String) -> anyhow::Result<()> {
+    let user_op = existing_storage_account_user_op(validate_rule, pm_rule).await;
+    validate(user_op)
+}
+
+#[tokio::test]
+async fn accept_plain_request() {
+    let context = get_global_context().await;
+    let (init_code, init_func) = create_opcode_factory_init_code("".to_string())
+        .await
+        .unwrap();
+    test_user_op(
+        "".to_string(),
+        None,
+        init_code,
+        init_func,
+        context.opcodes_factory.address,
+    )
+    .await
+    .expect("succeed");
+}
+
+#[tokio::test]
+#[should_panic]
```

Comment:
The `#[should_panic]` flag should be removed when the validating function is implemented. The case panic right now is because `validate` always return Ok